awswrangler/s3/_read.py

```diff
@@ -377,3 +378,20 @@ def _get_paths_for_glue_table(
                 )
 
     return paths, path_root, res
+
+
+def _get_num_output_blocks(
+    ray_args: RaySettings | None = None,
+) -> int:
+    ray_args = ray_args or {}
+    parallelism = ray_args.get("parallelism", -1)
+    override_num_blocks = ray_args.get("override_num_blocks")
+    if parallelism != -1:
```

Comment:
Shouldn't it raise an error instead of a warning. `override_num_blocks` does not accept -1 or does it? If not then it would raise an error too

Reply:
No - this is backwards-compatibility fix, same as in the ray code. If `parallelism` is passed and it's `> -1`, it will be routed to `override_num_blocks`. We will remove the warning and stop supporting `parallelism` arg in the next major release.